plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/functions/tablechanges/TableChangesSplitSource.java

```diff
@@ -49,84 +51,95 @@
 public class TableChangesSplitSource
         implements ConnectorSplitSource
 {
-    private final TrinoFileSystem fileSystem;
     private final String tableLocation;
-    private final AtomicLong currentVersion;
-    private final long tableReadVersion;
-    private final String transactionLogDir;
+    private final Iterator<ConnectorSplit> splits;
 
     public TableChangesSplitSource(
             ConnectorSession session,
             TrinoFileSystemFactory fileSystemFactory,
             TableChangesTableFunctionHandle functionHandle)
     {
         tableLocation = functionHandle.tableLocation();
-        fileSystem = fileSystemFactory.create(session);
-        currentVersion = new AtomicLong(functionHandle.firstReadVersion());
-        tableReadVersion = functionHandle.tableReadVersion();
-        transactionLogDir = getTransactionLogDir(tableLocation);
+        splits = prepareSplits(
+                functionHandle.firstReadVersion(),
```

Comment:
Can we have a test similar to `io.trino.plugin.deltalake.TestDeltaLakeConnectorTest#testReadCdfChanges` in the context of dealing with 

```
"query.schedule-split-batch-size", "1",
```

see `io.trino.plugin.hive.TestBucketedQueryWithManySplits`

With the default value of `1000` that we have by default for `query.schedule-split-batch-size`  I guess we're not currently actually testing this functionality with the current codebase (and your PR doesn't contain a corresponding test).